gsa/src/gmp/utils/calculator.js

```diff
@@ -0,0 +1,676 @@
+/* Copyright (C) 2017-2020 Greenbone Networks GmbH
+ *
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation, either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {isDefined} from 'gmp/utils/identity';
+
+const toFixed1 = value => {
+  let result = +(
+    Math.round(+(value.toString() + 'e' + 1)).toString() +
```

Comment:
Personally I consider toString() as an implementation detail and wouldn't use it directly. It is more like the __str__ method in Python which also doesn't get called directly. Either use string concatenation or template literals

```javascript
'' + foo

`${foo}` 
```

Reply:
I really did not want to use this function. :/
But all my own rounding Implementations didn't round the score as [expected](https://nvd.nist.gov/vuln-metrics/cvss/v2-calculator).